#!/usr/bin/env python
PACKAGE = "kuka_ros_zmq"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kuka_j0",    double_t,    0, "KUKA Joint 1", 0,  -170, 170)
gen.add("kuka_j1",    double_t,    0, "KUKA Joint 1", 0,  -120, 120)
gen.add("kuka_j2",    double_t,    0, "KUKA Joint 1", 0,  -170, 170)
gen.add("kuka_j3",    double_t,    0, "KUKA Joint 1", 0,  -120, 120)
gen.add("kuka_j4",    double_t,    0, "KUKA Joint 1", 0,  -170, 170)
gen.add("kuka_j5",    double_t,    0, "KUKA Joint 1", 0,  -120, 120)
gen.add("kuka_j6",    double_t,    0, "KUKA Joint 1", 0,  -175, 175)

size_enum = gen.enum([ gen.const("CartesianPP",      int_t, 0, "Cartesian PTP Motion"),
                       gen.const("CartesianSS",     int_t, 1, "Cartesian SmartServo"),
                       gen.const("CartesianDS",      int_t, 2, "Cartesian DirectServo"),
                       gen.const("JointbyJoint", int_t, 3, "Joint Space Control")],
                     "An enum to set mode")

gen.add("mode", int_t, 0, "KUKA Control Mode", 3, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "kuka_ros_zmq", "KUKAcontrol"))